### PR TITLE
Saturated Addition, Unsigned.

### PR DESCRIPTION
**Pull request by Scottbruceheart**

### CODE REVIEW

File: inc/zoo/swar/associative_iteration.h [hunk 1]
```diff
@@ -106,6 +106,36 @@ fullAddition(SWAR<NB, B> s1, SWAR<NB, B> s2) {
     return { result, BS{carry.value()}, BS{overflow.value()} };
 };
 
+static_assert( SWAR<4, u32>(0x1111'1111).value() == fullAddition(SWAR<4, u32>(0x0111'1101), SWAR<4, u32>(0x1000'0010)).result.value());
```
Comment (thecppzoo) [inline inc/zoo/swar/associative_iteration.h:109]: At this point of maturity of the library, I don't see the point in continuing to say `SWAR<4, uint32_t>` in the test files.  I think we can simply get away with defining however `using S4_32 = SWAR<4, uint32_t>`  we want type aliases globally in the test file; I advocate for the convention of `S` followed by the number of bits per lane, `_` and the number of bits in the unsigned integer we mean.
And we should remove `static_assert` from the user-facing headers.

File: inc/zoo/swar/associative_iteration.h [hunk 2]
```diff
@@ -106,6 +106,36 @@ fullAddition(SWAR<NB, B> s1, SWAR<NB, B> s2) {
     return { result, BS{carry.value()}, BS{overflow.value()} };
 };
 
+static_assert( SWAR<4, u32>(0x1111'1111).value() == fullAddition(SWAR<4, u32>(0x0111'1101), SWAR<4, u32>(0x1000'0010)).result.value());
+static_assert( SWAR<4, u32>(0x0000'0000).value() == fullAddition(SWAR<4, u32>(0x0111'1101), SWAR<4, u32>(0x1000'0010)).carry.value());
+static_assert( SWAR<4, u32>(0x0000'0000).value() == fullAddition(SWAR<4, u32>(0x0111'1101), SWAR<4, u32>(0x1000'0010)).overflow.value());
+
+template<int NB, typename B>
+constexpr ArithmeticResultTriplet<NB, B>
+fullSaturatedAddition(SWAR<NB, B> s1, SWAR<NB, B> s2) {
```
Comment (thecppzoo) [inline inc/zoo/swar/associative_iteration.h:115]: I think there are two flavors of saturation: unsigned and signed.

Comment (thecppzoo) [reply inc/zoo/swar/associative_iteration.h:115]: What about clamping?

File: inc/zoo/swar/associative_iteration.h (hunk 1 above)
Comment (Scottbruceheart) [reply inc/zoo/swar/associative_iteration.h:109]: SX_Y meaning bits per lane and bits in fundamental type is a good convention.

File: inc/zoo/swar/associative_iteration.h (hunk 2 above)
Comment (Scottbruceheart) [reply inc/zoo/swar/associative_iteration.h]: Name fixed, clamping is for a future PR.

**Comment by Scottbruceheart** [comment]
too diverged from main, making clean PR

**Pull request closed without merging**